src/app/service.rs

```diff
@@ -40,47 +39,28 @@ impl<T: Send + Sync> Service<http::Request<Incoming>> for AppService<T> {
     fn call(&self, request: http::Request<Incoming>) -> Self::Future {
         let mut params = PathParams::new(Vec::with_capacity(8));
         let mut next = Next::new(VecDeque::new());
+        let state = Arc::clone(&self.app.state);
 
-        let path = request.uri().path();
-
-        for binding in self.app.router.visit(path) {
-            for node in binding.results() {
-                let match_as_final = match (node.pattern(), binding.range()) {
-                    (Pattern::Dynamic(name), Some(range)) => {
-                        params.push(name.clone(), *range);
-                        binding.is_final()
-                    }
-                    (Pattern::Wildcard(name), None) => {
-                        params.push(name.clone(), [path.len(); 2]);
-                        true
-                    }
-                    (Pattern::Wildcard(name), Some([start, _])) => {
-                        params.push(name.clone(), [*start, path.len()]);
-                        true
-                    }
-                    _ => binding.is_final(),
-                };
-
-                if match_as_final {
-                    // Include the entire route stack of the matched node.
-                    next.extend(node.as_final().cloned());
-                } else {
-                    // Only include the middleware of the matched node.
-                    next.extend(node.as_partial().cloned());
+        for binding in self.app.router.visit(request.uri().path()) {
```

Comment:
The latest commit moves the responsibility of filtering route handlers from the API consumer of via-router into via-router. The result is a cleaner public API and likely better performance. Currently, wildcard path ranges are not accounted for. I'm going to make a change that fixes that after a quick lunch break.